internal/c/start.go

```diff
@@ -78,6 +115,29 @@ func (chSpec ChildSpec) DoStart(
 		// we cancel the childCtx on regular termination
 		defer cancelFn()
 
+		defer func() {
```

Comment:
Do you need to remove the `TODO:` above?

Reply:
Good catch